processor/src/main/java/org/mapstruct/ap/internal/model/GeneratedType.java

```diff
@@ -74,12 +74,17 @@ protected GeneratedType(TypeFactory typeFactory, String packageName, String name
         this.versionInformation = versionInformation;
         this.accessibility = accessibility;
 
-        this.generatedTypeAvailable = typeFactory.isTypeAvailable( "javax.annotation.Generated" );
-        if ( generatedTypeAvailable ) {
+        if ( typeFactory.isTypeAvailable( "javax.annotation.Generated" ) ) {
```

Comment:
Is this an 'or' situation? So when `java.annotation.Generated` is present than that should be used regardless of the presence of `java.annotation.processing.Generated`? Otherwise, the `if`.. `else if` should be flipped around.

Reply:
I had a better look at this. If we switch this then when you compile with Java 10 with Java 8 source then both types are present. Therefore I would propose to do something else. If `SourceVersion` is bigger than Java 8 we can use the `javax.annotation.processing.Generated`. Otherwise we use the `javax.annotation.Generated`.

Reply:
Ok.. that feels more logical.. Please go ahead when done with merge

Reply:
I will do a new PR as that would involve a bit more changes 😉